File: LeetCode/DataStruct/Python/Leetcode921-minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        l, r = 0, 0
        for c in s:
            if c == '(': l += 1
            elif c == ')':
                if not l: r += 1
                else: l -= 1
        return l + r

File: LeetCode/DataStruct/Rust/Leetcode921-minimum-add-to-make-parentheses-valid.rs
impl Solution {
  pub fn min_add_to_make_valid(s: String) -> i32 {
    let (s, mut l, mut r) = (s.as_bytes(), 0, 0);
    for &c in s.iter() {
      if c == b'(' {
        l += 1;
      } else if c == b')' {
        if l == 0 {
          r += 1;
        } else {
          l -= 1;
        }
      }
    }
    l + r
  }
}
